### PR TITLE
Remove references to `prop_index_site_code` view

### PR DESCRIPTION
**Pull request by esoergel**
@nickpell
I'm replacing these view calls with comparable SQL calls.  The tests compare the old and the new versions.  Once this is deployed, I'll remove the tests and the view itself.

### CODE REVIEW

File: corehq/apps/locations/forms.py [hunk 1]
```diff
@@ -16,9 +16,9 @@
 from corehq.apps.users.models import CommCareUser
 from corehq.apps.users.util import raw_username, user_display_string
 
-from .models import Location
+from .models import Location, SQLLocation
 from .signals import location_created, location_edited
-from .util import load_locs_json, allowed_child_types, lookup_by_property
+from .util import load_locs_json, allowed_child_types
```
Comment (dimagimon) [inline corehq/apps/locations/forms.py]: F401 'load_locs_json' imported but unused

File: corehq/apps/locations/tests/test_reupholster.py [hunk 2]
```diff
@@ -86,6 +86,54 @@ def test_filter_by_type_no_root(self):
...
+    # working for all locations: loc.by_site_code(loc.domain, loc.site_code)
+    result = Location.get_db().view(
+        'locations/prop_index_site_code',
+        reduce=False,
+        startkey=[domain, site_code.lower()],
+        endkey=[domain, site_code.lower(), {}],
+    ).first()
+    return Location.get(result['id']) if result else None
+
+def _sql_by_site_code(domain, site_code):
```
Comment (dimagimon) [inline corehq/apps/locations/tests/test_reupholster.py]: E302 expected 2 blank lines, found 1

**Comment by nickpell** [comment]
looks good, waiting on tests

**Comment by esoergel** [comment]
@nickpell tests passed

**Comment by dannyroberts** [comment]
good stuff!

**Pull request merged**